decorator_factory.py

```diff
@@ -0,0 +1,29 @@
+from functools import wraps
+
+
+class IncorrectTypeException(Exception):
+    """
+    Custom exception for comparing types.
+    """
+    def __init__(self, expected_type, actual_type):
+        self.message = f"Expected type {expected_type}, got {actual_type} instead"
+        super().__init__(self.message)
+
+
+def check_type(correct_type):
+    def check_type_decorator(f):
+        @wraps(f)
+        def wrapper(num):
```

Comment:
is it always a number? it can be a decorator for a function that receives a str.